test-data/unit/check-classes.test

```diff
@@ -894,6 +892,295 @@ a.f = a.f # E: Property "f" defined in "A" is read-only
 a.f.x # E: "int" has no attribute "x"
 [builtins fixtures/property.pyi]
 
+-- Descriptors
+-- -----------
+
+
+[case testAccessingNonDataDescriptor]
+class D:
+    def __get__(self, instance: Any, owner: Any) -> str:
+        return 'some string'
+
+class A:
+    f = D()
+
+a = A()
+reveal_type(a.f)  # E: Revealed type is 'builtins.str'
+
+[case testSettingNonDataDescriptor]
+class D:
+    def __init__(self, value: str) -> None:
+        self.value = value
+    def __get__(self, instance: Any, owner: Any) -> str:
+        return self.value
+
+class A:
+    f = D('some string')
+
+a = A()
+a.f = 'foo'
+a.f = D('different string')  # E: Incompatible types in assignment (expression has type "D", variable has type "str")
+
+
```

Comment:
Don't need double blank lines between tests. (In fact in tests we don't usually even put blank lines in the code at all, and we also often put small function bodies on the same line as the 'def'.)